### ISSUE TITLE
Delivery reports "coercing to Unicode: need string or buffer, NoneType found"

### ISSUE THREAD

**Issue by dleehr** [Duke-GCB/D4S2#98]
Client sent a deliver and reported the following error

```
$ ddsclient deliver -p REMOVED --user removed --skip-copy --msg-file text.txt --resend
Delivering project.
 
coercing to Unicode: need string or buffer, NoneType found
```

Server logs indicate a 200 status code:

```
"POST /api/v1/deliveries/NN/send/ HTTP/1.1" 200 7541 "-" "python-requests/2.18.4"
```

And the admin record does show an email sent.

I think the issue is that the associated DukeDSUser record had an empty name. It is a service account that has no name registered in DukeDS.

**Comment by johnbradley** [comment]
I think this might be related to #96. DukeDSClient doesn't make use of the response after sending a delivery. It does however print out the email address sent to. I think in this case we fixed the email within D4S2 but it was still wrong in DukeDS.

https://github.com/Duke-GCB/DukeDSClient/blob/4f05340facf52ddc5dc41351d7e3f6ff5fc2e5b2/ddsc/ddsclient.py#L329
```
            print("Delivery email message sent to " + dest_email)
```

If the user we are sending to's email field is blank we should prevent delivery in DukeDSClient.

I checked and the setup for the user in question has been fixed on the DukeDS side but there are still users with `null` email addresses.
